src/Preg.php

```diff
@@ -14,67 +14,164 @@
 class Preg
 {
     /**
-     * @param string $pattern
-     * @param string $subject
-     * @param array<string|null> &$matches
-     * @param int    $flags PREG_UNMATCHED_AS_NULL, only available on PHP 7.2+
-     * @param int    $offset
-     * @return int Number of matches
+     * @param string|string[] $pattern
+     * @param string|string[] $replacement
+     * @param string|string[] $subject
+     * @param int             $limit
+     * @param int             $count Set by method
+     * @return string|string[]
      */
-    public static function match($pattern, $subject, &$matches = null, $flags = 0, $offset = 0)
+    public static function filter($pattern, $replacement, $subject, $limit, &$count)
     {
-        $result = Regex::match($pattern, $subject, $flags, $offset);
-        $matches = $result->matches;
+        $error = null;
+        set_error_handler(function ($errno, $errstr) use (&$error) {
+            $error = $errstr;
+            return true;
```

Comment:
Probably the most transparent would be to return false and call the previous error handler if any. Essentially we would just hook in to know if a warning happens or not

Reply:
I'm leaning towards completely removing array $subjects (it makes things too complex), which in this case means that we can simply drop the error_handler, because `null` is returned.